sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3944,8 +3944,8 @@ object functions {
       | * @since 2.3.0
       | */
       |def udf(f: UDF$i[$extTypeArgs], returnType: DataType): UserDefinedFunction = {
-      |  val func = f$anyCast.call($anyParams)
-      |  SparkUserDefinedFunction($funcCall, returnType, inputSchemas = Seq.fill($i)(None))
+      |  val func = $funcCall
+      |  SparkUserDefinedFunction(func, returnType, inputSchemas = Seq.fill($i)(None))
```

Comment:
There is another place:

https://github.com/apache/spark/blob/8a17d26784c53fb50b6373b566aab71135c8956f/sql/core/src/main/scala/org/apache/spark/sql/UDFRegistration.scala#L720

Although it claims to `Register a deterministic Java UDF0 instance...`, maybe returning same value not an issue, but it does call at driver...

Reply:
ah, let me fix it too while I am here.